### PR TITLE
Fix: Use flow numbers to map jobs to tasks in 8.1.x workflows.

### LINKED ISSUE #5247: review: Mapping Task Jobs back to Tasks
**Describe the bug**

This bug is visible in the current cylc review. Previously run jobs are missing.

**Steps to reproduce the bug**

Run a workflow like the one below:

```
[scheduling]
    [[graph]]
        R1 = foo

[runtime]
    [[foo]]
        script = """
            if [[ $CYLC_TASK_TRY_NUMBER -eq 1 ]]; then
                cylc message -- moo1
                false
            elif [[ $CYLC_TASK_TRY_NUMBER -eq 2 ]]; then
                cylc message -- moo2
                false
            else
                cylc message -- moo3
            fi
        """
        execution retry delays = 5*PT3S
```

**Expected behavior**
Three jobs should be visible, as shown below, only 3/3 is visible.

**Screenshots**

image

**Additional context**

In the Cylc8 database, we have two tables which are currently joined to produce the information necessary to display jobs in cylc review. 

The two tables involved in this join are `task_states` and `task_jobs`.

These two tables store the following info: 
image image

It is important to note that `task_states` only stores information about the task, not info about previously run jobs. Note also that flow_nums are not held in the `task_jobs` table.

Prior to https://github.com/cylc/cylc-flow/pull/5140, we joined these two tables on `cycle` and `name`.

However, consider the workflow below:

```
[scheduling]
    [[graph]]
        R1 = """trigger-task => a => b => c => d => e => f => g => h
        """
[runtime]
    [[trigger-task]]
    script = """
            cylc trigger --flow=new trigger-demo-new//1/d
            cylc__job__poll_grep_workflow_log -E '1/d.*started'
        """
    [[a, b, c]]
        script = sleep 1;
    [[d, e, f, g, h]]
        script = sleep 1;
```

This workflow has a new 'flow'. Joining only on `cycle` and `name` produced incorrect duplicate entries in cylc review. We changed the join by adding `submit_num` (from task_jobs table), correcting this problem but creating the problem of missing jobs detailed above. 

There is no join which will cater for both of these problems. We could flip the join but we have information lost which we need from jobs 1 & 2 which is not held in `task_states` table.

Note: although this is a current cylc review bug. The real issue is mapping task jobs back to tasks (and associated lost information), which will be an issue for the replacement.

Having spoken to @oliver-sanders, we believe adding `flow_nums` to the `task_jobs` table, and changing the join to include flow_nums, should solve the problem. 

This is slightly messy as may cause back-compat problems with cylc review. 
 

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

### LINKED ISSUE #5249: tests/f: job-submission/03 failing on DB upgrade
This test `tests/functional/job-submission/03-job-nn-remote-host-with-shared-fs.t` is failing on master due to a DB  upgrade issue.

```
sqlite3.OperationalError: duplicate column name: is_manual_submit
```

Odd thing is it doesn't appear to have `is_manual_submit` in the DB before. Is this a problem with the upgrader or the test DB?

#5187

**Pull requests welcome!**
This is an Open Source project - please consider contributing a bug fix
yourself (please read `CONTRIBUTING.md` before starting any work though).

### PR DESCRIPTION
**Pull request by wxtim**
Closes #5247

- Allow Cylc Review to see all jobs with different flow nums for Cylc 8.1 workflows
- Warn that not all jobs for different flows will be visible for Cylc 8.0 workflows

## Checklist

- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Manually tested - it's legacy software
- [x] `CHANGES.md` entry included if this is a change that can affect users

### CODE REVIEW

**Comment by wxtim** [comment]
@datamel 
- Please select a second reviewer
- Note that I have test workflows for you to look at using you example from the issue at `myaccount/x2/run1` and `run2`.

**Review by datamel — COMMENTED** [review]
Problems are fixed for me. Super thanks for the demo workflows. I have un-reproducible traceback which can be seen in my workflows:`trigger-demo-new/run1`.Think it is related to it being an old stale one, maybe worth a bit of defensive code? 
A slightly separate issue which I noticed which may or may not be worth fixing is the  `job #` file has 1 of 1, think this should be 1 of 2? ....
image

File: lib/cylc/review.py [hunk 1]
```diff
@@ -379,9 +379,14 @@ def taskjobs(
...
-        entries, of_n_entries = self.suite_dao.get_suite_job_entries(
-            user, suite, cycles, tasks, task_status, job_status, order,
-            per_page, (page - 1) * per_page)
+        (
+            entries, of_n_entries, eight_point_zero
+        ) = self.suite_dao.get_suite_job_entries(
+            user, suite, cycles, tasks,
+            task_status, job_status, order,
+            per_page, (page - 1) * per_page
+        )
```
[the remark this answers is not in the document]
Comment (datamel) [reply lib/cylc/review.py:388]: Resolving this as it had a corrupted db, which I think was the cause and is unlikely to happen.

File: lib/cylc/cylc-review/static/css/cylc-review.css [hunk 2]
```diff
@@ -113,3 +113,8 @@ th,
 .livestamp {
   white-space: nowrap;
 }
+
+.warning {
+  background-color: #FFBF00;
+  border-radius: 4px;
+}
```
[the remark this answers is not in the document]
Comment (oliver-sanders) [reply lib/cylc/cylc-review/static/css/cylc-review.css:120]: FYI: You should be able to configure VSCode to use Unix file line end convention.

**Review by MetRonnie — APPROVED** [review]
Tested out

**Pull request merged**